cloudscale/resource_cloudscale_volume_test.go

```diff
@@ -0,0 +1,270 @@
+						"cloudscale_volume.basic", "server_uuids.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudscaleVolume_Detach(t *testing.T) {
+	var server cloudscale.Server
+	var volume cloudscale.Volume
+
+	rInt1 := acctest.RandInt()
+	rInt2 := acctest.RandInt()
+
+	serverConfig := testAccCheckCloudscaleServerConfig_basic(rInt1)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudscaleVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: serverConfig + "\n" + volumeConfig_attached(rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleServerExists("cloudscale_server.basic", &server),
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					resource.TestCheckResourceAttr(
+						"cloudscale_volume.basic", "server_uuids.#", "1"),
+					assertVolumeAttached(&server, &volume),
+				),
+			},
+			{
+				Config: serverConfig + "\n" + volumeConfig_detached(rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					resource.TestCheckResourceAttr(
+						"cloudscale_volume.basic", "server_uuids.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudscaleVolume_Reattach(t *testing.T) {
+	var server1, server2 cloudscale.Server
+	var volume cloudscale.Volume
+
+	rInt1 := acctest.RandInt()
+	rInt2 := acctest.RandInt()
+	rInt3 := acctest.RandInt()
+
+	serverConfig := testAccCheckCloudscaleServerConfig_basic(rInt1)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudscaleVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: serverConfig + "\n" + volumeConfig_attached(rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleServerExists("cloudscale_server.basic", &server1),
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					resource.TestCheckResourceAttr(
+						"cloudscale_volume.basic", "server_uuids.#", "1"),
+					assertVolumeAttached(&server1, &volume),
+				),
+			},
+			{
+				Config: serverConfig + "\n" + volumeConfig_reattached_volume(rInt3, rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleServerExists("cloudscale_server.reattach_server", &server2),
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					assertVolumeAttached(&server2, &volume),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudscaleVolumeDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudscale.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudscale_volume" {
+			continue
+		}
+
+		id := rs.Primary.ID
+
+		// Try to find the volume
+		_, err := client.Volumes.Get(context.Background(), id)
+		if err != nil && !strings.Contains(err.Error(), "Not found") {
```

Comment:
Could you use the HTTP status code here instead of checking if the error message contains a string?